spam2 stable issues, UI update and modals added

keshavsethi opened the pull request:
There were few bugs in the UI of the spam management dashboard. 
- Wrong ids in /comments and /revisions 
- created modal for comment and revisions 
- default order nodes wrt time 
- **UI is updated** as there were few issues related to banner in stable. Apart from that this new UI requires very less external css and it is based on bootstrap mostly. it has everything which was in old UI. This is fully responsive and better than the previous UI. I have attached some screenshots and gifs.
![Screenshot from 2020-06-20 18-14-00](https://user-images.githubusercontent.com/36025262/85202196-4bc9d580-b322-11ea-928f-ce5d10ebef8c.png)
![Screenshot from 2020-06-20 18-10-04](https://user-images.githubusercontent.com/36025262/85202197-4ec4c600-b322-11ea-9f27-1080c2e1fdbc.png)
![Screenshot from 2020-06-20 18-09-47](https://user-images.githubusercontent.com/36025262/85202199-51272000-b322-11ea-9a7d-ea99927cc95e.png)

There are a few issues that I am facing currently: 
1. /spam2 and /spam2/revision are giving error in stable while wiki and comment are fine, I think it might be due to invalid ID or something. I have changed it but not sure if it is the only issue. 
I have added (&.) instead of (.) it will discard all values where node.author is nill. Please review this.
` <a href="/profile/<%= node.author&.name %>"><%= node.author&.name %> <%= node.author&.new_contributor%></a> `
Please review @jywarren @SidharthBansal @cesswairimu @pydevsg @ananyaarun @VladimirMikulic @Uzay-G @emilyashley 
Thanks!!

VladimirMikulic commented:
Fix CI :)

jywarren commented:
This is a follow-up to #7961 -- right? Would you mind linking together the issues in a central place so we can see them all in relation? Thank you!

jywarren commented:
Hi @keshavsethi -- here, can you see this? https://sentry.io/share/issue/0d9664f6f2db4194978b7671837928ee/

Also were you able to log into Sentry and do searches and link like I've done in the previous comment? Thanks!

jywarren commented:
Hmm, regarding anniversary banner, is there a way to not use absolute positioning but to have it push down below if the banner exists? Otherwise, yes, we could exempt it. Thank you!!!

jywarren commented:
and finally, sorry, would you be able to add a little more descriptive title here? it really helps when trying to link issues together with `#` autocompletion. Thank you!!!!!

keshavsethi commented:
> This is a follow-up to #7961 -- right? Would you mind linking together the issues in a central place so we can see them all in relation? Thank you!

#7961(closed)was the first draft then I /spam2 was changed a lot. This PR is just a few updates to that including errors in the stable. Thanks!

keshavsethi commented:
> Hi @keshavsethi -- here, can you see this? https://sentry.io/share/issue/0d9664f6f2db4194978b7671837928ee/
> 
> Also were you able to log into Sentry and do searches and link like I've done in the previous comment? Thanks!

@jywarren Yes, I saw error before in sentry. It was due to  `data: { confirm: 'Are you sure you want to delete "'+node.path+'"?' }`  I have changed node.path here in this PR to #{node.path}. I hope it should work now. Thanks!

keshavsethi commented:
> Hmm, regarding anniversary banner, is there a way to not use absolute positioning but to have it push down below if the banner exists? Otherwise, yes, we could exempt it. Thank you!!!

@jywarren I have added margin like this 
```
<% cache('feature_anniversary-banner', skip_digest: true) do %>
  <style>
  .nav_top{
   margin-top:48px;
  }
  .sidenav{
    margin-top:48px;
  }
  </style>
<% end %>
```
It was working fine locally and tests were also fine but not sure in the stable. Can you please verify? Thanks!

keshavsethi commented:
> and finally, sorry, would you be able to add a little more descriptive title here? it really helps when trying to link issues together with `#` autocompletion. Thank you!!!!!

Sure, Thanks I will write more descriptive titles in the future.

cesswairimu commented on app/controllers/spam2_controller.rb:
```diff
@@ -1,18 +1,18 @@
-  before_action :require_user, only: %i(spam spam_revisions mark_comment_spam publish_comment spam_comments)
+  before_action :require_user, only: %i(_spam _spam_revisions _spam_comments)
 
   def _spam
```
Hey @keshavsethi , not a big deal but just wanted to know why you choose to use ` _methodname`

keshavsethi replied:
spam is method name for /spam(admin controller) while "_spam" is method name for /spam2 (spam2 controller). There was an error in method name so I corrected it

cesswairimu replied:
gotcha :+1:

keshavsethi commented:
@jywarren @SidharthBansal @cesswairimu @pydevsg @ananyaarun @VladimirMikulic @Uzay-G @emilyashley I have changed UI  a bit. Please review.
Thanks!!

pydevsg commented on app/assets/stylesheets/spam2.css:
```diff
@@ -36,10 +36,6 @@
 	cursor: pointer;
 }
 
-.alert {
```
hey @keshav234156 , inplace of alert what else have you added .
Also can you tell me if you have added something in CSS when a model pop ups, thanks.

keshavsethi replied:
Alert is there, it is just a z-index for that which I have added when there was another top nav for bulk moderation(previous UI). Now I have moved that nav below the card and there is no need to specify z-index for alert. I have used a modal bootstrap class here.

pydevsg replied:
:+1:

keshavsethi commented:
I have added a specific column search in moderation nav.it will help moderators to filter node to a greater extent. Refer to following screenshot
![Screenshot from 2020-06-18 19-13-05](https://user-images.githubusercontent.com/36025262/85029356-d6cb9400-b199-11ea-94e2-9e4fa1b93ee5.png)

jywarren commented:
![85029356-d6cb9400-b199-11ea-94e2-9e4fa1b93ee5](https://user-images.githubusercontent.com/24359/85426711-d429bf80-b548-11ea-82ed-c7332b85b41b.png)

Hi, this is super awesome. I have a few suggestions before we merge -- some easy, some maybe for a follow-up?

1. (A) - Could we have a “?” button too leading to /wiki/moderation#Dashboard where we add some documentation of each button? 
2. (B and C) - could we group these together better? I don't know if Bootstrap has a style for this, but adding "Filter by author" instead of just "Author" could help clarify, and grouping them together could help people understand that the input text field is the term that is being used for the filtering. Also, if it has a placeholder text (via Bootstrap styles), it could say "Filter text" or something?
3. (D) perhaps after it says "filter" we should have a `(clear)` link that allows us to clear filters from that spot. 

Thanks, what do you think? 🎉

ebarry commented:
Thanks @keshavsethi, this dashboard is looking good!
👍 to @jywarren's 3 points. 
I"m also wondering if there could be text "Select content below to take any of the following actions" above the buttons, and if the buttons could be shown in paler colors until there is content selected?

keshavsethi commented:
> ![85029356-d6cb9400-b199-11ea-94e2-9e4fa1b93ee5](https://user-images.githubusercontent.com/24359/85426711-d429bf80-b548-11ea-82ed-c7332b85b41b.png)
> 
> Hi, this is super awesome. I have a few suggestions before we merge -- some easy, some may be for a follow-up?
> 
> 1. (A) - Could we have a “?” button too leading to /wiki/moderation#Dashboard where we add some documentation of each button?
> 2. (B and C) - could we group these together better? I don't know if Bootstrap has a style for this, but adding "Filter by author" instead of just "Author" could help clarify, and grouping them together could help people understand that the input text field is the term that is being used for the filtering. Also, if it has a placeholder text (via Bootstrap styles), it could say "Filter text" or something?
> 3. (D) perhaps after it says "filter" we should have a `(clear)` link that allows us to clear filters from that spot.
> 
> Thanks, what do you think?

Hey @jywarren, I have added an info button, I will link it with wiki page once it is made. I have made B and C as a part of a single button group and added a tooltip for it for reference. There is a reset button in the header of the table which becomes red if we apply any filter and black if no filter is applied I have also disabled bulk moderation buttons if nothing is selected. and I have updated UI a bit. please refer to the following Screenshot. 
![Screenshot from 2020-06-24 00-56-02](https://user-images.githubusercontent.com/36025262/85450441-674d0e80-b5b6-11ea-99f0-9ed217f7a8c5.png)
@jywarren Please review these changes, Thanks!!

keshavsethi commented:
> Thanks @keshavsethi, this dashboard is looking good!
>  to @jywarren's 3 points.
> I"m also wondering if there could be text "Select content below to take any of the following actions" above the buttons, and if the buttons could be shown in paler colors until there is content selected?

@ebarry, I have disabled the buttons if nothing is selected, Thanks!!
![Screenshot from 2020-06-24 00-56-02](https://user-images.githubusercontent.com/36025262/85450998-0a058d00-b5b7-11ea-8edd-e0daf6930577.png)

jywarren commented:
Oh WOW. @keshavsethi this is very fast work! I'll merge it now!

jywarren commented:
It'll be testable at https://stable.publiclab.org/spam2 soon. And if that works, i may try to push it into production today! We'll keep an eye out!!!

jywarren commented:
Congrats!! 🎉

keshavsethi commented:
> It'll be testable at https://stable.publiclab.org/spam2 soon. And if that works, i may try to push it into production today! We'll keep an eye out!!!

Sure, Thanks!! :smile:

cesswairimu commented:
:tada: :tada:
